lua_examples/net_send_queue.lua

```diff
@@ -0,0 +1,46 @@
+------------------------------------------------------------------------------
+-- Net send queueing helper
+--
+-- See also
+--   https://nodemcu.readthedocs.org/en/dev/en/modules/net/#netsocketsend
+--
+-- Based on Vladimir Dronnikov's
+--   MQTT queuing publish helper
+--   https://github.com/dvv/nodemcu-thingies/blob/master/mqtt-queue-helper.lua
+--   LICENCE: http://opensource.org/licenses/MIT
+--
+-- Send arbitrary payload:
+--   local socket = net.createConnection(net.TCP, 0)
+--   socket:connect(port, ip)
+--   local net_send = dofile("net_send_queue.lua")(socket)
+--   net_send(payload1)
+--   net_send(payload2)
+--   net_send(payload3)
+--
+------------------------------------------------------------------------------
+do
+  -- cache
+  local shift = table.remove
+  -- queue handler
+  local make_sender = function(socket)
+    local queue = { }
+    local is_sending = false
+    local function send()
+      if #queue > 0 then
+        local item = shift(queue, 1)
```

Comment:
More importantly if we are doing this then its worth doing a lookahead marshal of any queued records something like:
``` Lua
local total_bytes, ready = 0, {}
while #queue>0 do
  total_bytes = total_bytes + #(queue[1])
  if (total_bytes> 1460) then break end
  ready[#ready + 1] = shift(queue, 1)
end
socket:send(table.concat(ready)), send) -- this is safe as long as records have been split in chunks < 1460
```